angular2-jwt.ts

```diff
@@ -109,7 +109,19 @@ export class AuthHttp {
       this.setGlobalHeaders(this.config.globalHeaders, providedOpts);
     }
 
-    newOptions = newOptions.merge(new RequestOptions(providedOpts));
+    let providedOptions = new RequestOptions(providedOpts);
+    if (newOptions.headers && newOptions.headers.keys().length > 0) {
```

Comment:
checking keys().length is not needed. The forEach won't do anything if there are no keys